soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -77,6 +77,9 @@ __imr void z_mp_entry(void)
 {
 	cpu_early_init();
 
+	if (prid() != start_rec.cpu)
+		k_cpu_idle();
```

Comment:
Experimentally it is for all the intel_adsp devices.  But at the hardware design level it can be anything.  ESP32 had funny numbers in that register.  Worst-case is that we'd have to shift over to _current_cpu->id if this changes.

Reply:
if statement needs braces